enhancements/installer/ibmcloud-ipi.md

```diff
@@ -0,0 +1,181 @@
+---
+title: openshift-ipi-on-ibmcloud
+authors:
+  - "@jeffnowicki"
+  - "@BobbyRadford"
+reviewers:
+  - @staebler
+approvers:
+  - @staebler
+creation-date: 2021-05-03
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# OpenShift Installer Provisioned Infrastructure (IPI) on IBM Cloud
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes adding support for OpenShift 4 Installer Provisioned Infrastructure (IPI) on IBM Cloud VPC (Gen 2) infrastructure. It describes the necessary enhancements, tooling and documentation to enable this capability.
+
+## Motivation
+
+Users expect OpenShift to be available through multiple cloud providers. Customers with high-availability and enhanced security requirements may wish to take advantage of [IBM Cloud VPC (Gen 2) infrastructure](https://www.ibm.com/cloud/vpc). Leveraging IPI, the process of installing OpenShift on IBM Cloud can be simplified.
+
+
+### Goals
+
+The primary goal of IBM Cloud IPI is to provide users with an easier path to running OpenShift 4 on VPC infrastructure in IBM Cloud data centers.
+
+To achieve that goal, we will follow the pattern of other IPI supported platforms:
+- Enhance the installer to survey customer for IBM Cloud options.
+- Prepare IBM Cloud Terraform (TF) module for control plane provisioning. The [IBM Cloud Terraform Provider](https://github.com/IBM-Cloud/terraform-provider-ibm) will utilized.
+- Integrate with IBM Cloud VPC machine API provider. The [IBM Cloud Cluster API Provider](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud) will be utilized.
+- Integrate with IBM Cloud Controller Manager (CCM) and enhance requisite operators to enable cluster functionality on IBM Cloud.
+- Provide IBM Cloud IPI user documentation [here](https://github.com/openshift/installer/tree/master/docs/user/ibmcloud/).
+- Provide CI artifacts required to test the IBM Cloud IPI installer.
+
+
+### Non-Goals
+
+None at this time.
+
+## Proposal
+
+- Implement installer CLI prompts in order to build a default/minimal `install-config.yaml` for IBM Cloud :
+  ```shell
+  ? SSH Public Key /Users/someone/.ssh/id_rsa.pub
+  ? Platform ibmcloud
+  ? Resource Group ID default (34ffb674f7c4466398dcd257a0dac58e)
+  ? Region us-south
+  ? RHCOS Custom Image rhcos-ibmcloud-470
```

Comment:
You could add that as a non-goal for this enhancement to make it clear that it doesn't block this work but is related.

Reply:
We found a means of automating the creation of custom rhcos images on behalf of users, so the rhcos image name is no longer a prompted survey question requiring input from users.

Reply:
Excellent. It would be good to capture that history in the "alternatives" section of this doc.

Is the image going to be an optional input? I don't know what we do about images on other platforms, so I'm not sure what would be consistent.